Debugger Memory Search: add Array of byte type

solid1353 opened the pull request:
### Description of Changes
Added `Array of byte` option to debugger memory search + validation for empty strings

### Rationale behind Changes
- `Array of byte` option was added to allow searching for byte sequences more easily, without bothering with endianness. For example, if you want to find a chunk of bytes from game's source files which is directly loaded into memory and stored there, you can just copy-paste, without using CheatEngine or whatever. CE is now also harder to use due to PS2 memory not having fixed addresses anymore. It should work both with delimiting spaces and without them
- As for validation, if you search for an empty string at the moment, it searches infinitely with memory usage skyrocketing instantly

### Suggested Testing Steps
I've done some regress testing for older options and edge-cases for new ones as well as normal scenarios, but nothing too extensive. I guess smoke-testing all the options should be enough

RedDevilus commented:
You can still use EEmem as a variable in Cheat Engine to get something similar to the old fixed addresses as it dynamically points where it needs to be just as a reminder.

solid1353 commented:
> You can still use EEmem as a variable in Cheat Engine to get something similar to the old fixed addresses as it dynamically points where it needs to be just as a reminder.

Yeah, I know, it's just that having an option in debugger is more convenient than going back and forth with CE

F0bes reviewed (commented):
I'm thinking it'll be a better idea to change the search string function to search bytes. That way it'll make sense for both string and array of byte types, instead of the need to convert the bytes to a string.

solid1353 commented:
> I'm thinking it'll be a better idea to change the search string function to search bytes. That way it'll make sense for both string and array of byte types, instead of the need to convert the bytes to a string.

Agreed. Had exactly the same idea, but didn't want to change too much without feedback, since it's my first PR. I'll update the solution then

refractionpcsx2 commented:
Fobes' suggestion sounds like a good idea to me :) You've plenty of time to work on this, there's no rush!

solid1353 commented:
So I've changed some typings and removed some conversions, but maybe that's not quite what you wanted? `DebugInterface` doesn't seem to allow for anything other than sequential byte reads in this case. Also is it fine to use Qt typings in `searchWorkerByteArray` or do I need to convert `QByteArray` to `vector` or `char*` beforehand?

F0bes reviewed (approved):
Tested it and it worked for me

F0bes commented:
> Also is it fine to use Qt typings in `searchWorkerByteArray` or do I need to convert `QByteArray` to `vector` or `char*` beforehand?

Yes, as long as it is part of the `pcsx2-qt` target.

stenzek reviewed (approved):
Code changes LGTM also
